core/src/test/java/io/substrait/type/proto/ConsistentPartitionWindowRelRoundtripTest.java

```diff
@@ -36,7 +36,12 @@ void consistentPartitionWindowRoundtrip() {
                         .declaration(windowFunctionDeclaration)
                         // lead(a)
                         .arguments(Arrays.asList(b.fieldReference(input, 0)))
-                        .options(Collections.emptyMap())
+                        .options(
+                            Arrays.asList(
+                                FunctionOption.builder()
+                                    .name("option")
+                                    .addValues("VALUE1", "VALUE2")
+                                    .build()))
```

Comment:
meta: The [lead function](https://github.com/substrait-io/substrait/blob/7dbbf0468083d932a61b9c720700bd6083558fa9/extensions/functions_arithmetic.yaml#L1750) doesn't actually define any options. Potentially, we should reject both:
* Building plans w/ invalid options.
* Reading protobuf plans in w/ invalid options.

I think this is outside the scope of this PR.

Reply:
Fair 😅 but yep agreed it'd be a separate PR